### PR TITLE
Fixes motion alarms

### PR DESCRIPTION
**Pull request by Cyberboss**
Roast me

:cl: Cyberboss
fix: Fixed motion alarms instantly triggering and de/un/re/triggering
/:cl:

The relevant changes are in motion.dm and ai_monitored.dm. Everything else is a replacement of /area/turret_protected -> /area/ai_monitored/turret_protected which is why this PR is a bitch.

### CODE REVIEW

File: code/game/area/ai_monitored.dm [hunk 1]
```diff
@@ -1,24 +1,29 @@
...
+			return
 
 /area/ai_monitored/Exited(atom/movable/O)
-	if (ismob(O) && motioncamera)
-		motioncamera.lostTarget(O)
-
-
+	..()
+	if (ismob(O) && motioncameras.len)
+		world << "[src] calling exit for [O]"
```
Comment (Shadowlight213) [inline code/game/area/ai_monitored.dm]: Honk

**Review by Shadowlight213 — CHANGES_REQUESTED** [review]

File: code/game/area/ai_monitored.dm (hunk 1 above)
Comment (CyberbossJHCB) [inline code/game/area/ai_monitored.dm]: merp

File: code/game/machinery/camera/motion.dm [hunk 2]
```diff
@@ -15,28 +15,32 @@
...
-		for (var/mob/target in motionTargets)
-			if (target.stat == 2) lostTarget(target)
-			// If not detecting with motion camera...
-			if (!area_motion)
-				// See if the camera is still in range
-				if(!in_range(src, target))
-					// If they aren't in range, lose the target.
-					lostTarget(target)
+		for (var/mob/target in getTargetList())
+			if (target.stat == 2 || \
```
Comment (phil235) [inline code/game/machinery/camera/motion.dm]: `target.stat == DEAD`, use the defines.

File: code/game/area/ai_monitored.dm [hunk 3]
```diff
@@ -1,24 +1,29 @@
...
-	var/obj/machinery/camera/motioncamera = null
+	var/list/obj/machinery/camera/motioncameras = list()
+	var/list/motionTargets = list()
 
-
-/area/ai_monitored/New()
+/area/ai_monitored/initialize()
 	..()
-	// locate and store the motioncamera
-	spawn (20) // spawn on a delay to let turfs/objs load
```
Comment (phil235) [inline code/game/area/ai_monitored.dm]: you tested that removing that spawn doesn't cause any issues?

Comment (CyberbossJHCB) [inline code/game/area/ai_monitored.dm]: It was basically a ghetto initialize function. So moving the actions from New to initialize handles things fine. 

I actually had a stack_trace in there that detected monitored areas without motion cameras and it actually detected that the cameras in AI atmos and AI cyborg service are NOT motion cameras. And that was just boxstation. If you want I could add it back in.

**Pull request merged**